src/storage-client/src/storage_collections.rs

```diff
@@ -329,6 +335,13 @@ pub struct StorageCollectionsImpl<
     /// all of the controllers and Coordinator.
     envd_epoch: NonZeroI64,
 
+    /// Whether or not this [StorageCollections] is in read-only mode.
+    ///
+    /// When in read-only mode, we are allowed to affect changes to external
+    /// systems, including, for example, acquiring and downgrading critical
+    /// [SinceHandles](SinceHandle)
+    read_only: Arc<AtomicBool>,
```

Comment:
Is there a short explanation of why this is an atomic in an Arc instead of a plain value?

Reply:
Basically all of the `StorageCollectionsImpl` state is shared, via `Arcs`, so this one has to be as well.